### PR TITLE
Rid Yieldmo bidder of OpenrtbBidder.

### PR DESCRIPTION
**Pull request by CTMBNara**

### CODE REVIEW

File: src/main/java/org/prebid/server/bidder/yieldmo/YieldmoBidder.java [hunk 1]
```diff
@@ -1,52 +1,130 @@
...
+import java.util.Collections;
 import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.stream.Collectors;
 
-public class YieldmoBidder extends OpenrtbBidder<ExtImpYieldmo> {
+public class YieldmoBidder implements Bidder<BidRequest> {
+
+    private static final JsonPointer EXT_DATA_PBADSLOT_POINTER_EXPRESSION = JsonPointer.valueOf("/data/pbadslot");
```
Comment (snahornyi) [inline src/main/java/org/prebid/server/bidder/yieldmo/YieldmoBidder.java:38]: Pls rename to PBADSLOT_POINTER

File: src/main/java/org/prebid/server/bidder/yieldmo/YieldmoBidder.java [hunk 2]
```diff
@@ -1,52 +1,130 @@
...
-        final JsonNode pbadslotNode = imp.getExt().at("/data/pbadslot");
-        final String gpid = !pbadslotNode.isMissingNode()
-                ? StringUtils.defaultIfEmpty(pbadslotNode.asText(), null)
-                : null;
+        if (modifiedImps.isEmpty()) {
+            return Result.withErrors(errors);
+        }
+
+        final BidRequest modifiedRequest = bidRequest.toBuilder().imp(modifiedImps).build();
+        return Result.of(Collections.singletonList(makeRequest(modifiedRequest)), errors);
```
Comment (snahornyi) [inline src/main/java/org/prebid/server/bidder/yieldmo/YieldmoBidder.java:70]: Minor. Pls \n before return statement if method contains more than one line. Also pls check similar

File: src/main/java/org/prebid/server/bidder/yieldmo/YieldmoBidder.java [hunk 3]
```diff
@@ -1,52 +1,130 @@
...
-            modifiedImp.ext(mapper.mapper().valueToTree(YieldmoImpExt.of(impExt.getPlacementId(), gpid)));
+            return mapper.mapper().convertValue(imp.getExt(), YIELDMO_EXT_TYPE_REFERENCE).getBidder();
         } catch (IllegalArgumentException e) {
             throw new PreBidException(e.getMessage(), e);
         }
+    }
+
+    private Imp modifyImp(Imp imp, ExtImpYieldmo ext) {
+        final JsonNode pbadslotNode = imp.getExt().at(EXT_DATA_PBADSLOT_POINTER_EXPRESSION);
+        final String gpid = StringUtils.defaultIfBlank(pbadslotNode.asText(), null);
```
Comment (snahornyi) [inline src/main/java/org/prebid/server/bidder/yieldmo/YieldmoBidder.java:83]: For easier readability let's check that pbadslotNode is not MissingNode than use asText() method

File: src/main/java/org/prebid/server/bidder/yieldmo/YieldmoBidder.java [hunk 4]
```diff
@@ -1,52 +1,130 @@
...
+    }
+
+    private static List<BidderBid> extractBids(BidRequest bidRequest, BidResponse bidResponse) {
+        if (bidResponse == null || CollectionUtils.isEmpty(bidResponse.getSeatbid())) {
+            return Collections.emptyList();
         }
-        return BidType.video;
+
+        final String currency = bidResponse.getCur();
+        final Map<String, BidType> impIdToBidType = bidRequest.getImp().stream()
```
Comment (snahornyi) [inline src/main/java/org/prebid/server/bidder/yieldmo/YieldmoBidder.java:114]: For now let's update resolveBidType method to similar in project

File: src/main/java/org/prebid/server/bidder/yieldmo/YieldmoBidder.java [hunk 5]
```diff
@@ -1,52 +1,130 @@
...
+        }
+    }
+
+    private static List<BidderBid> extractBids(BidRequest bidRequest, BidResponse bidResponse) {
+        if (bidResponse == null || CollectionUtils.isEmpty(bidResponse.getSeatbid())) {
+            return Collections.emptyList();
         }
-        return BidType.video;
+
+        final String currency = bidResponse.getCur();
```
Comment (snahornyi) [inline src/main/java/org/prebid/server/bidder/yieldmo/YieldmoBidder.java:113]: Can be moved directly in BidderBid.of() method

**Review by snahornyi — CHANGES_REQUESTED** [review]

File: src/main/java/org/prebid/server/bidder/yieldmo/YieldmoBidder.java (hunk 3 above)
Comment (CTMBNara) [reply src/main/java/org/prebid/server/bidder/yieldmo/YieldmoBidder.java:83]: Do you wanna smth like
```
final String gpid = !pbadslotNode.isMissingNode()
                ? StringUtils.defaultIfBlank(pbadslotNode.asText(), null)
                : null;
```

**Review by snahornyi — APPROVED** [review]

**Pull request merged**